File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 2/21/2026, 9:44:01 AM
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int n=nums.size();
        
        for(int i=0;i<n-1;)
        {
            int j=i,k;
            k=j;
            while(i<n-1 && nums[i]==0)
            {
              i++;  
            }
            while( j>=0 && nums[j]==0)
            {
                j--;
            }
            if(k!=j)
            {j=j+1;}
            if(i!=j)
                
            {nums[j]=nums[i];
                nums[i]=0;
            }
            else
            {
                i++;
            }
        }
        
    
    }
};

File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 2/21/2026, 9:43:58 AM
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        pos=0
        i=0
        n=len(nums)
        while i<n and pos<n:
            if nums[i]!=0:
                nums[pos]=nums[i]
                if pos!=i:
                    nums[i]=0
                pos+=1
            i+=1